src/fb.coffee

```diff
@@ -115,7 +122,28 @@ class FBMessenger extends Adapter
             @robot.emit "fb_richMsg", envelope
             @_processAttachment event, envelope, attachment for attachment in envelope.attachments
         if event.message.text?
-            @receive new TextMessage envelope.user, event.message.text
+            text = @cleanMessageText event.message.text, envelope.room
+            msg = new TextMessage envelope.user, text, event.message.mid
+            @receive msg
+            @robot.logger.info "Reply message to room/message: " + envelope.user.name + "/" + event.message.mid
+
+    cleanMessageText: (text, chat_id) ->
+        # if AUTOHEAR is not turned on, don't clean
+        if !process.env['FB_AUTOHEAR']
+            return text
+        
+        # If it is a private chat, automatically prepend the bot name if it does not exist already.
+        if (chat_id > 0)
+            # Strip out the stuff we don't need.
+            text = text.replace(/^\//g, '').trim()
```

Comment:
an option to let the bot pick up user input due to the way in which people talk to bot in Facebook (direct messages). This is inspired by hubot-telegram adapter. People normally don't start every line with "bot-name give me the weather". The function cleans it and append the bot name in front.

For cross-adapter compatibility it's important to let platform like Slack have the distinction between `hear` and `respond` since it's channel heavy.